### PR TITLE
refactor(meta): extract file_format.proto and stage.proto.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

I am working on https://github.com/datafuselabs/databend/issues/10376.

it will make codes much more clear to do this refactoring before adding new structures (about file formats) .
and it is easier to review in a separate pr.

nothing except `paths of PB messages in code level` should be changed.

format-related Messages were defined inside StageInfo,  they are refactored to the top level.

not PB tags affected,  so should not cause any compatibility problem.

### CODE REVIEW

File: src/meta/protos/proto/file_format.proto [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  Gzip = 1;
+  Bz2 = 2;
+  Brotli = 3;
+  Zstd = 4;
+  Deflate = 5;
+  RawDeflate = 6;
+  Lzo = 7;
+  Snappy = 8;
+  // We used to assign `None` as 9, it's a mistake.
+  // Please change this value to 0 instead in next version bump.
```
Comment (ariesdevil) [inline src/meta/protos/proto/file_format.proto:42]: Should this be modified in this PR?

**Review by ariesdevil — APPROVED** [review]
Codes LGTM, it is recommended to describe in detail the compatibility involved.

**Comment by youngsofun** [comment]
> Codes LGTM, it is recommended to describe in detail the compatibility involved.

not PB tags affected, so should not cause any compatibility problem.

**Review by drmingdrmer — APPROVED** [review]
Looks like there are just file moves. LGTM!

**Pull request merged**